src/pages/SwapPage/Swap.tsx

```diff
@@ -38,6 +58,67 @@ function Swap({ swap }: SwapProps) {
   const [alphaTradeAction, betaTradeAction] =
     properties.role === Role.Alice ? tradeActions : tradeActions.reverse();
 
+  const communicationActions = (swap.actions || []).filter(
+    action => action.name === "accept" || action.name === "decline"
+  );
+
+  const ledgerActions = (swap.actions || []).filter(
+    action => action.name !== "accept" && action.name !== "decline"
+  );
+  const alphaLedgerActions =
+    properties.role === Role.Alice
+      ? ledgerActions.filter(
+          action =>
+            action.name === "deploy" ||
+            action.name === "fund" ||
+            action.name === "refund"
+        )
+      : ledgerActions.filter(action => action.name === "redeem");
+  const betaLedgerActions =
+    properties.role === Role.Alice
+      ? ledgerActions.filter(action => action.name === "redeem")
+      : ledgerActions.filter(
+          action =>
+            action.name === "deploy" ||
+            action.name === "fund" ||
+            action.name === "refund"
+        );
+
+  const [
+    {
+      state: {
+        actionExecutionStatus,
+        activeLedgerActionDialog,
+        activeSirenParameterDialog
+      },
+      sideEffect
+    },
+    dispatch
+  ] = useReducer(reducer, initialState);
+
+  useEffect(() => {
```

Comment:
This is now duplicated right? Can we extract that into a custom hook? (i.e. just extract the function and name it something starting with `use`.